### DISCUSSION TITLE
Deployment issue: Did you forget to export a function or a server?

### DISCUSSION THREAD

**Discussion by aecorredor** [vercel/vercel#4045] [General]
I'm deploying a node server, and I think I've encountered an error on my side, but it's been hard to track because apparently the lambda is crashing and the only error I get is this:

```
2020-04-10T17:40:34.189Z	a006e262-c205-46ae-aa47-46e04c50bc93	ERROR	
No exports found in module "./packages/server/dist/src/index.js".2020-04-10T17:40:34.189Z	

a006e262-c205-46ae-aa47-46e04c50bc93	ERROR	Did you forget to export a function or a server?
RequestId: a006e262-c205-46ae-aa47-46e04c50bc93 Error: Runtime exited with error: exit status 1Runtime.ExitError
```

Every subsequent request for the next 15 minutes (which is how long the crashed lambda stays alive for from what I understand) I get the same error.

Also, the first error I got was a 10s timeout: `Task timed out after 10.01 seconds`, which I understand is from the hobby limit plan. My question is, when that timeout is reached does the node process in the lambda crash? if so, why?

I'm trying to debug where my issue could be, but the logs are not helping me too much in this case.

Thanks in advance for any help.

EDIT: Also, I'm seeing the lambda stay down for more than 15 minutes, my subsequent requests are not going through at all. So I'm not sure how the timing is working. Keep in mind, this same deployment was working fine until I hit the endpoint that caused the error. It's a GraphQL apollo server with express, so I expose a `/graphql` endpoint plus a few restful routes.

**Comment by aecorredor** [comment]
What I've done for now is to add a 5 second query timeout to PostgreSQL. I'm using sequelize to access the DB and apparently one of the queries was failing and just stalling, which consequently made the lambda execute for more than 10s and thus reach the limit. I'll report back if that was the issue.

Either way, I wanted to get an insight into exactly how crashes for processes inside a lambda are handled. I know each lambda lives for up to 15 minutes, so does that means that if a lambda crashes any future requests to this lambda for those 15 minutes will fail? Or does the lambda just get killed as soon as the process inside it crashes?

**Comment by lorstenoplo** [comment]
I am experiencing the exact same issue,
GitHub repo : https://github.com/lorstenoplo/eccom-api

**Reply by aecorredor** [reply]
This ended up being an issue on my end. I had some transactions where I missed passing the actual `{ transaction: t }` options object to certain sequelize calls that required it. They were just a little deeper within the call stack. This caused the operation to stall and go over the 10 second limit.

**Reply by lorstenoplo** [reply]
Yeah I am getting that my async function is not able to listen within the time limit, it is taking more than 60s according to heroku.
What I wanted is just some help with my code because I don't see ( or I am not able to ) any such mistakes that is making it slow.
I would thankful if anyone could look into my code and help me.
